File: 프로그래머스/2/131127. 할인 행사/할인 행사.py
def solution(want, number, discount):
    answer = 0
    answerdict = {}
    
    for i in range(len(want)) :
        answerdict[want[i]] = number[i]
    
    
    for i in range(0,len(discount)-10+1):
        discountarr = discount[i:i+10] 
        
        cntdict = answerdict.copy()
        
        for product in discountarr :
            if product in cntdict.keys():
                cntdict[product]-=1         
        
        zerocnt = True
        
        for i in cntdict.values():
            if i !=0 :
                zerocnt = False
                break
        if zerocnt == True:
            answer+=1
        
    return answer

File: 프로그래머스/2/131127. 할인 행사/할인 행사.java
import java.util.*;

class Solution {
    public int solution(String[] want, int[] number, String[] discount) {
        int answer = 0;
        
        Map<String, Integer> wantmap = new HashMap<>();
        
        int total = 0;
        for (int i = 0 ; i<number.length; i++)
        {
            wantmap.put(want[i],number[i]);
            total+= number[i];
        }
        
        int day = 0 ;
        while (day <= (discount.length-total))
        {
            Map<String,Integer> comparemap = new HashMap<>();
            for (int i = day; i<total+day; i++)
            {
                comparemap.put(discount[i],comparemap.getOrDefault(discount[i],0)+1);
            }
            boolean flag = true;
            for (String key : comparemap.keySet())
            {
                if (!wantmap.keySet().contains(key))
                {
                    flag= false;
                    break;
                }
                
                if (comparemap.get(key) != wantmap.get(key))
                {
                    flag= false;
                    break;
                }
            }
            
            if (flag)
            {
                answer+=1;
            }
            day+=1;
        }
 
        
        return answer;
    }
}